Formula/retroshare.rb

```diff
@@ -0,0 +1,48 @@
+class Retroshare < Formula
+  desc "Friend to friend communication platform"
+  homepage "https://retroshare.cc/"
+  url "https://github.com/RetroShare/RetroShare/releases/download/v0.6.5/RetroShare-v0.6.5-source-with-submodules.tar.gz"
+  sha256 "d974d25cbcceccaf6e23e3b8d683d61e91b08c1853b3be0ecb06281809979cb3"
+  head "https://github.com/RetroShare/RetroShare.git"
+
+  depends_on "pkg-config" => :build
+  depends_on "miniupnpc"
+  depends_on "openssl"
+  depends_on "qt"
+  depends_on "rapidjson"
+  depends_on "sqlcipher"
+  depends_on "xapian"
+
+  def install
+    # Avoid RetroShare build system set wrong SDK version
+    inreplace "retroshare.pri",
+        "macx:CONFIG *= rs_macos10.11", ""
+    inreplace "retroshare-gui/src/retroshare-gui.pro",
+        "	target.path = \"$${BIN_DIR}\"",
+        "	target.path=#{prefix}"
+    system "qmake",
+        "PREFIX=#{prefix}",
+        "CONFIG+=c++11", "CONFIG+=no_retroshare_nogui", "CONFIG+=no_libresapi",
+        "CONFIG+=no_libresapihttpserver", "CONFIG+=rs_deep_search",
+        "RS_MAJOR_VERSION=0", "RS_MINOR_VERSION=6", "RS_MINI_VERSION=5",
+        "RS_EXTRA_VERSION=-macOS-#{MacOS.version}-homebrew",
+        "INCLUDEPATH+=#{Formula["miniupnpc"].opt_include}",
+        "INCLUDEPATH+=#{Formula["openssl"].opt_include}",
+        "INCLUDEPATH+=#{Formula["rapidjson"].opt_include}",
+        "INCLUDEPATH+=#{Formula["sqlcipher"].opt_include}",
+        "INCLUDEPATH+=#{Formula["xapian"].opt_include}",
+        "QMAKE_LIBDIR+=#{Formula["miniupnpc"].opt_lib}",
+        "QMAKE_LIBDIR+=#{Formula["openssl"].opt_lib}",
+        "QMAKE_LIBDIR+=#{Formula["rapidjson"].opt_lib}",
+        "QMAKE_LIBDIR+=#{Formula["sqlcipher"].opt_lib}",
+        "QMAKE_LIBDIR+=#{Formula["xapian"].opt_lib}",
+        "DATA_DIR=#{share}"
+    system "make"
+    system "make", "install"
+    bin.install_symlink "#{prefix}/retroshare.app/Contents/MacOS/retroshare"
+  end
+
+  test do
+    puts shell_output("find #{prefix}")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I read that part of the documentation too, but the executable run the GUI with every option, even --help...